### DISCUSSION TITLE
A Counterexample to John's Principle About Secrets?

### DISCUSSION THREAD

**Discussion by gregfowlerphd** [JisooSeo/101-WG-Discussions#16] [General]
In our working group meeting a couple weeks ago, John (@johnbeve) went through a talk he'd given on the ontology of secrets. During the Q&A, I presented what I took to be a counterexample to a principle John presented, and John responded. Here I’d like to further discuss this (purported) counterexample and explain why I’m not convinced by John’s response.

The principle in question, which I’ve given a name for ease of reference, appears on slides 27 and 28 of John's PowerPoint:

> **Secret Revelation and Prevention (SRP):** If a secret keeper recognizes a situation in which the secret will be revealed and the secret keeper can prevent revelation, then they must actively prevent it.

My (purported) counterexample involved situations in which a secret keeper doesn’t prevent revelation even though they can because (i) doing so will make them appear as though they’re trying to keep a secret and (ii) their desire not to appear that way outweighs their desire to prevent revelation.

In response, John said that situations of this sort don’t constitute a counterexample to (SRP) because in such situations, the secret keeper decides to give up the secret. Hence, they are no longer a secret keeper, in which case (SRP)’s antecedent is false.

I don’t think John’s response is successful. Here’s why. While John is correct that the secret keeper decides to give up the secret in the relevant situations, this happens only after a deliberative process, which takes time (however minimal). Prior to the completion of this process, they’re still a secret keeper and (SRP)’s antecedent is true. However, its consequent is false. Hence, (SRP) is false.

(Note that (SRP) has both modal (“must”) and temporal (“will”) elements, and it might be worth developing a formulation that makes them explicit. I've decided against doing so here, but we may need to do so in the comments if there’s disagreement over my reply to John’s response.)

Even if I’m right, there might still be ways to salvage something like (SRP). One possibility would be to include a clause explicitly noting the possibility of giving up the secret:

> **Secret Revelation and Prevention, Revised 1 (SRPR1):** If a secret keeper recognizes a situation in which the secret will be revealed, can prevent revelation, and doesn’t decide to give up the secret, then they must actively prevent revelation.

(Or, equivalently: If a secret keeper recognizes a situation in which the secret will be revealed and the secret keeper can prevent revelation, then they must either decide to give up the secret or actively prevent revelation.)

However, (SRPR1) doesn’t seem to tell us as much, or at least as much of interest, about what it is to be/keep a secret as (SRP), so it’s worth exploring other possibilities. One such possibility would be to appeal to the notion of a disposition:

> **Secret Revelation and Prevention, Revised 2 (SRPR2):** If a secret keeper recognizes a situation in which the secret will be revealed and the secret keeper can prevent revelation, then they have a disposition to actively prevent revelation.

This seems more interesting than (SRPR1), and while I’m not sure it’s quite right, I don’t think it’s subject to the sort of counterexample to (SRP) I described.

**Comment by johnbeve** [comment]
Good thoughts Greg, as usual. 

Not represented entirely in the slides, but the principle I had in mind is more like:
 
- **Secret Revelation and Prevention (SRP3)**: If the holder H of a secret S wants to keep S a secret & believes there is a situation C which will obtain in the future where S is to be revealed to the whomever it is withheld from & believes they can prevent C from occurring, then H must attempt to prevent C from occurring. 

If H doesn't attempt to prevent C from occurring, then either (1) H doesn't want to keep S a secret any longer, (2) H doesn't believe C will reveal S, (3) H doesn't believe they can prevent C from occurring, or some combination. 

Re 1: In your example, S doesn't want to keep S a secret any longer because doing so is outweighed by other desires.
Re 2: You might also describe a situation so that H isn't sure that C will reveal S, worries that behaving to prevent the reveal will look suspicious, and so is willing to take the risk of it being revealed by not acting. 

I think we're actually not far from one another ;)

**Comment by gregfowlerphd** [comment]
@johnbeve:

Thanks for responding, John. :) And sorry not to reply sooner; some life stuff got in the way.

I appreciate you spelling out SRP3. I _think_ I might still have concerns about it based on the sort of situation I discussed in my post, but I'm not sure. Like SRP, SRP3 has modal and temporal elements, and making these explicit might make it easier to evaluate both it and purported counterexamples to it. Here's an attempt to do so:

> Necessarily, for all times t, if (i) at t, the holder H of a secret S wants to keep S a secret, (ii) at t, H believes that at some time t'>t, a situation C will obtain where S is revealed to whomever it is withheld from, and (iii) at t, H believes they can prevent C from occurring, then (iv) at some time t', H attempts to prevent C from occurring.

Does this seem to capture the intended meaning of SRP3?

**Comment by gregfowlerphd** [comment]
@johnbeve: Just getting a chance to come back to this topic.

So, in your reply, you presented a principle you called SRP3,* which you presented as underlying the principle you put forward in your presentation on secrets. SRP3 went like this:

> **Secret Revelation and Prevention (SRP3):** If the holder H of a secret S wants to keep S a secret & believes there is a situation C which will obtain in the future where S is to be revealed to the whomever it is withheld from & believes they can prevent C from occurring, then H must attempt to prevent C from occurring.

In response, I noted that SRP3 has modal and temporal elements and that evaluating both it and potential counterexamples to it might be easier if these elements were made explicit, and I stated a version of SRP3 intended to do so:

> **Secret Revelation and Prevention, Explicit (SRP3E):** Necessarily, for all times t, if (i) at t, the holder H of a secret S wants to keep S a secret, (ii) at t, H believes that at some time t'>t, a situation C will obtain where S is revealed to whomever it is withheld from, and (iii) at t, H believes they can prevent C from occurring, then (iv) at some time t', H attempts to prevent C from occurring.

While I'm still not entirely certain whether SRP3E captures the intended meaning of SRP3, here's why it matters. I think scenarios of the sort discussed in my initial post may constitute counterexamples to SRP3E. Remember that these are scenarios in which, roughly, a secret keeper in which a secret keeper doesn’t prevent the revelation of a secret because they're worried about appearing as though they're trying to keep a secret.

To see why, consider the temporal structure of such a scenario: At a certain time t1, the holder H of a secret S wants to keep S a secret but comes to believe that at some time t>t1, a situation C will obtain where S is revealed to whomever it is withheld from. H then considers whether they can prevent C from occurring. At t2, H comes to believe that they can prevent C from occurring. However, after further consideration, at t3, H realizes that every avenue available for preventing C from occurring will make them appear as though they're trying to keep a secret. H then--perhaps even later, at t4, after H has weighed their desire to keep S a secret against their desire not to appear as though they're keeping a secret--stops wanting to keep S a secret any longer because doing so is outweighed by their other aforementioned desire. As a result, at no time t does H attempt to prevent C from occurring.

Notice that while it is true that at the end of the scenario (t3 or t4, depending), H does not want to keep S a secret and hence clause (i) of SRP3E is false, this isn't true at every time in the scenario. In particular, consider a time t* between t2 and t3. At t*, H still wants to keep S a secret (since H doesn't give up that desire until later), so clause (i) is true. Moreover, at t*, H believes that at some later time t, a situation C will obtain where S is revealed to whomever it is withheld from, making (ii) true as well. Finally, t* satisfies (iii) as well: at t*, H believes they can prevent C from occurring. Thus, every conjunct of the antecedent of SRP3E is true of t*. However, its consequent is false, since H never attempts to prevent C from occurring. It looks like we've got ourselves a counterexample!

As I noted in a recent email, I think there are other scenarios that might cause problems for SRP3E (and other versions of your original principle). In particular, I mentioned scenarios involving "time-limited" secrets, such as the salient secret involved in throwing a surprise party. I'll try to write something up about that soon, but if you have the time, I'm curious to hear what you think of the preceding. :)


*Though you might have intended to call it SRPR3, in keeping with the names of the revised principles in my initial post. Regardless, I'll keep calling it SRP3.

**Comment by gregfowlerphd** [comment]
@johnbeve: Just coming back to this...

I mentioned previously that I think your principle might run into problems in cases involving "time-limited" secrets.

Consider a case in which Bert and the rest of Ernie's friends are planning to throw him a surprise party, which they've scheduled to start at 8 PM on 10/1/2024.

It seems like in this case, at every time between when they did the scheduling and 8 PM on 10/1/2024, Bert is a holder of a secret S--namely, that Ernie's friends have planned a surprise party for him at 8 PM on 10/1/2024--that he wants to keep secret (from Ernie). Moreover, it seems that at every one of those times, Bert believes that at some later time (8 PM on 10/1/2024) a situation C will obtain in which S is revealed to Ernie--in particular, Ernie's arrival at the party at 8 PM on 10/1/2024. Moreover, it might be that at some of these times, Bert believes he can prevent C from occurring (by, say, cancelling or rescheduling the party).

Thus, it might be that every conjunct of the antecedent of SRP3E is true in this case. However, it needn't be the case that Bert actually makes any attempt to prevent C from occurring, so the consequent need not be true. Hence, the case constitutes a counterexample to SRP3E (and, I think, to other versions of your principle).

And I believe structurally similar cases might arise in other cases involving "time-limited" secrets. Of particular interest here are cases involving a national security/intelligence community context, given the automatic declassification of certain documents after a set amount of time.
